backend/src/v1/data/repositories/wordRepository.ts

```diff
@@ -1,36 +1,80 @@
 import { Word } from "../models/models";
-
-interface WordRepository {
-    getWords(filterParams: {}): Word[];
-    getWord(id: string): Word;
-    addWord(data: AddWordData): void;
-    updateWord(word: Word): void;
-    deleteWord(id: string): void;
+import { db } from "../../../db/database";
+import { DeleteResult, InsertResult } from "kysely";
+export interface WordRepository {
+    getWords(languageCode?: string): Promise<Word[]>;
+    getWord(id: number): Promise<Word>;
+    insertWord(data: WordInsert): Promise<{ insertedRows: string; }>;
+    updateWord(data: WordUpdate): Promise<{ updatedRows: string; }>;
+    deleteWord(id: number): Promise<{ deletedRowsCount: string; }>;
 };
 
 const WordRepositoryImpl: WordRepository = {
-    getWords: function (filterParams: {}): Word[] {
-        throw new Error("Function not implemented.");
+    getWords: async (languageCode: string): Promise<Word[]> => {
+        let query = db.selectFrom("word").selectAll();
+        if (languageCode) {
+            query.where("language_code", "=", languageCode);
+        }
+        let result = await query.execute();
+        return result.map((entity) => {
+            return {
+                id: entity.word_id,
+                text: entity.word_text,
+                languageCode: entity.language_code
+            };
+        });
     },
-    getWord: function (id: string): Word {
-        throw new Error("Function not implemented.");
+    getWord: async (id: number): Promise<Word> => {
+        let data = await db.selectFrom("word")
+            .selectAll()
+            .where("word.word_id", "=", id)
+            .executeTakeFirstOrThrow();
+
+        return {
+            id: data?.word_id,
+            text: data?.word_text,
+            languageCode: data?.language_code
+        };
     },
-    addWord: function (data: AddWordData): void {
-        throw new Error("Function not implemented.");
+    insertWord: async (data: WordInsert): Promise<{ insertedRows: string; }> => {
+        let query = db.insertInto("word").values({
+            word_text: data.text,
+            language_code: data.languageCode
+        });
+        let result: InsertResult = await query.executeTakeFirstOrThrow();
```

Comment:
You don't need an explicit type annotation here (drop the `: InsertResult`). Same for `DeleteResult` below. Generally you don't want to write out a type that TypeScript can figure out on its own.